src/Microsoft.DotNet.Interactive.Formatting/TabularDataFormatter.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright(c).NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Interactive.Formatting
+{
+    public static class TabularDataFormatter
+    {
+        // https://specs.frictionlessdata.io/table-schema/#language
+        public const string MimeType = "application/table-schema+json";
+
+        internal static ITypeFormatter[] DefaultFormatters { get; } = DefaultTabularDataFormatterSet.DefaultFormatters;
+
+
+        public static TabularDataSet ToTabularDataSet(this IEnumerable source)
```

Comment:
This method and its return type should probably be internal.

The type of `source` should probably be `IEnumerable<T>` .This doesn't exactly enforce homogeneous contents (which are necessary for valid tabular formatting) but it helps.